persistent/Database/Persist/Class/PersistStore.hs

```diff
@@ -48,6 +49,14 @@ class (HasPersistBackend backend) => IsPersistBackend backend where
     -- to accidentally run a write query against a read-only database.
     mkPersistBackend :: BaseBackend backend -> backend
 
+-- | This class witnesses that two backend are compatible, and that you can
+-- convert from the @sub@ backend into the @sup@ backend. This is similar
+-- to the 'HasPersistBackend' and 'IsPersistBackend' classes, but where you
+-- don't want to fix the type associated with the 'PersistEntityBackend' of
+-- a record.
+class BackendCompatible sup sub where
```

Comment:
This class provides "wiggle room" for other backends to be supported by eg Esqueleto, as long as they can be converted to a SQL backend.